src/query/storages/system/src/tables_table.rs

```diff
@@ -283,3 +310,28 @@ where TablesTable<T>: HistoryAware
         AsyncOneBlockSystemTable::create(TablesTable::<T> { table_info })
     }
 }
+
+pub fn find_eq_db_database(expr: &Expr<String>, visitor: &mut impl FnMut(&str, &Scalar)) {
```

Comment:
this function is similar to `find_eq_db_table`, you can reuse the function

Reply:
Yes, refactor it into utils.